### PR TITLE
registration

### PR DESCRIPTION
**Pull request by kevin-wise-infusionsoft**
Registration working inside of CAS

### CODE REVIEW

File: cas/src/main/resources/templates/casExpiredPassView.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout" xmlns:th="http://www.thymeleaf.org" layout:decorate="../layouts/login">
+<html xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout" xmlns:th="http://www.thymeleaf.org" layout:decorate="../layouts/infusionsoft">
```
Comment (bbooth) [inline cas/src/main/resources/templates/casExpiredPassView.html]: Why change this?  The login layout is used for anything in the login flow.

File: cas/src/main/resources/templates/casLoginView.html [hunk 2]
```diff
@@ -1,9 +1,24 @@
 <!DOCTYPE html>
-<html xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout" xmlns:th="http://www.thymeleaf.org" layout:decorate="../layouts/login">
+<html xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout" xmlns:th="http://www.thymeleaf.org" layout:decorate="../layouts/infusionsoft">
```
Comment (bbooth) [inline cas/src/main/resources/templates/casLoginView.html]: Why change this? The login layout is used for anything in the login flow.

File: cas/src/main/java/org/apereo/cas/infusionsoft/config/InfusionsoftCasConfiguration.java [hunk 3]
```diff
@@ -36,6 +38,14 @@
 public class InfusionsoftCasConfiguration implements AuthenticationEventExecutionPlanConfigurer {
 
     @Autowired
+    @Lazy
```
Comment (kevin-wise-infusionsoft) [inline cas/src/main/java/org/apereo/cas/infusionsoft/config/InfusionsoftCasConfiguration.java]: If I remember right, there was a circular dependency, so without the lazy it wouldn't start up.

File: cas/src/main/java/org/apereo/cas/infusionsoft/domain/User.java [hunk 4]
```diff
@@ -104,6 +105,15 @@ public void setPasswordRecoveryCodeCreatedTime(DateTime passwordRecoveryCodeCrea
         this.passwordRecoveryCodeCreatedTime = passwordRecoveryCodeCreatedTime;
     }
 
+    @Column(name = "ip_address", length = 32, nullable = true)
```
Comment (kevin-wise-infusionsoft) [inline cas/src/main/java/org/apereo/cas/infusionsoft/domain/User.java]: It was added in the old CAS at the same time as the EULA changes. It appears to be attempting to capture the IP address of the user at registration time.

File: cas/src/main/java/org/apereo/cas/infusionsoft/config/properties/AccountCentralConfigurationProperties.java [hunk 5]
```diff
@@ -5,30 +5,6 @@
 import org.springframework.boot.context.properties.NestedConfigurationProperty;
 
 @ConfigurationProperties("accountCentral")
-public class AccountCentralConfigurationProperties {
+public class AccountCentralConfigurationProperties extends HostConfigurationProperties {
```
Comment (kevin-wise-infusionsoft) [inline cas/src/main/java/org/apereo/cas/infusionsoft/config/properties/AccountCentralConfigurationProperties.java]: Registration path and url were no longer used, so host was the only property left.  I was just trying to clean it up and make it consistent with the config entries for other services (CRM, CustomerHub, Community, Marketplace).  Extending it allows us to add more properties.  At the time we hadn't deployed to proofing, so it didn't seem like a big deal. Would you like me to change it back?

File: cas/src/main/resources/templates/casExpiredPassView.html (hunk 1 above)
Comment (kevin-wise-infusionsoft) [inline cas/src/main/resources/templates/casExpiredPassView.html]: The only two layouts in the old CAS were the login flow and the Account Central pages. So of course the registration pages used the AC pages. But now that they are in CAS, they have the same basic requirements as the login pages.  So, I was renaming it to make it clear that it's not just used for the login page anymore.

File: cas/src/main/java/org/apereo/cas/infusionsoft/web/controllers/AuthenticateController.java [hunk 6]
```diff
@@ -114,18 +111,13 @@ public ResponseEntity authenticateUser(String username, String password, String
             if (error == null) {
                 UserDTO userDTO = new UserDTO(loginResult.getUser(), userService.findActiveUserAccounts(loginResult.getUser()), appHelper);
 
-                // TODO: This is a hack to make it lowercase.  Remove when mobile app does case-insensitive comparisons
```
Comment (kevin-wise-infusionsoft) [inline cas/src/main/java/org/apereo/cas/infusionsoft/web/controllers/AuthenticateController.java]: Yes, although based on our discussion yesterday we might have to worry about the Outlook plugin. Maybe we can treat this as a v2 api, and the v1 api continues to live in Account Central. Once an app cuts over to the new URL, they would have to be case-insensitive.  What do you think?

File: cas/src/main/java/org/apereo/cas/infusionsoft/config/InfusionsoftCasConfiguration.java (hunk 3 above)
Comment (bbooth) [inline cas/src/main/java/org/apereo/cas/infusionsoft/config/InfusionsoftCasConfiguration.java]: That should not be the fix.  We should not have circular dependencies.  I will pull it down and see what is going on.

**Pull request merged**